packages/analysis-report/README.md

```diff
@@ -0,0 +1,585 @@
+
+#### Marker Implementation
+
+A marker can be either a function or an array of marking instructions:
+
+```javascript
+// Function marker
+const functionMarker = () => {
+  return [
+    {
+      marked: "This text needs improvement",
+      original: "This text needs improvement",
+      position: {
+        startOffset: 0,
+        endOffset: 24
+      }
+    }
+  ];
+};
+
+// Array marker
+const arrayMarker = [
+  {
+    _startOffset: 0,
+    _endOffset: 24,
+    _marked: "This text needs improvement",
+    _original: "This text needs improvement"
+  }
+];
+```
+
+#### Marking Flow
+
+1. User clicks the eye icon (mark button)
+2. `onMarkButtonClick` is called with:
+   - `id`: The unique identifier of the assessment result
+   - `marker`: The marking instructions
+3. The parent application should then:
+   - Apply the marks to the text
+   - Update the UI to show the marked text
+   - Handle mark toggling (removing marks when clicked again)
+
+### Edit Button System
+
+The edit button system facilitates navigation to the relevant editing field for a specific assessment result.
+
+```javascript
+<ContentAnalysis
+  {...results}
+  onEditButtonClick={(id, fieldName) => {
+    // fieldName could be "title", "metaDescription", "text", etc.
+    focusField(fieldName);
+  }}
+/>
+```
+
+#### Edit Button Behavior
+
+1. **Visibility**: Edit buttons only appear when:
+   - `hasEditButton` is true for the result
+   - A valid `editFieldName` is provided
+   - The result is not suppressed
+
+2. **Navigation**: When clicked, should:
+   - Focus the corresponding input field
+   - Scroll to the field if needed
+   - Optionally highlight the field
+
+3. **Accessibility**: 
+   - Includes proper ARIA labels (e.g., "Edit your meta description")
+   - Supports keyboard navigation
+   - Provides visual feedback on focus/hover
+
+### Integration Example
+
+```jsx
+function SEOAnalysisWithMarking({ content }) {
+  const [activeMarker, setActiveMarker] = useState("");
+  const [marks, setMarks] = useState([]);
+
+  const handleMarkClick = (id, marker) => {
+    if (activeMarker === id) {
+      // Remove marks
+      setMarks([]);
+      setActiveMarker("");
+    } else {
+      // Apply new marks
+      const newMarks = typeof marker === "function" 
+        ? marker() 
+        : marker;
+      setMarks(newMarks);
+      setActiveMarker(id);
+    }
+  };
+
+  const handleEditClick = (id, fieldName) => {
+    const field = document.querySelector(`[name="${fieldName}"]`);
+    if (field) {
+      field.focus();
+      field.scrollIntoView({ behavior: "smooth" });
+    }
+  };
+
+  return (
+    <>
+      <ContentAnalysis
+        {...results}
+        marksButtonActivatedResult={activeMarker}
+        onMarkButtonClick={handleMarkClick}
+        onEditButtonClick={handleEditClick}
+      />
+      <TextEditor
+        content={content}
+        marks={marks}
+        onMarkingRequired={handleMarkClick}
+      />
+    </>
+  );
+}
+```
+
+## Components Reference
```

Comment:
This section is pretty much a duplicate of the `Core Components` section, except from the component descriptions that are slightly more detailed here.